[E 240506 11:04:24 views:154] API request failed: 'NoneType' object has no attribute 'lower'
    Traceback (most recent call last):
      File "/Users/arshil.khan/Desktop/stock_comparison_project/stock_comparison_app/views.py", line 90, in generate_excel
        interval = request.POST.get('interval').lower()
    AttributeError: 'NoneType' object has no attribute 'lower'
[E 240506 11:05:16 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-04-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:05:16 views:116] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:12:17 views:264] Error processing comparison data: name 'exchange_token' is not defined
    Traceback (most recent call last):
      File "/Users/arshil.khan/Desktop/stock_comparison_project/stock_comparison_app/views.py", line 225, in comparison_form
        angel_data = fetch_angel_data(stock_name, from_date, to_date, interval)
      File "/Users/arshil.khan/Desktop/stock_comparison_project/stock_comparison_app/views.py", line 186, in fetch_angel_data
        "symboltoken": exchange_token,
    NameError: name 'exchange_token' is not defined
[E 240506 11:16:27 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:16:28 views:116] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:17:34 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:17:34 views:116] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:19:52 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:19:52 views:116] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:26:05 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-06 00:00', 'todate': '2024-05-07 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:26:05 views:118] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:28:38 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:28:38 views:118] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:30:25 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:30:25 views:119] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:31:15 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:31:15 views:120] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:32:11 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'fromdate': '2024-05-01 00:00', 'todate': '2024-05-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:32:12 views:121] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1020","message":"Interval accepts one of (1minute,30minute,day,week,month)","propertyPath":"getHistoricalCandleData.interval","invalidValue":"None","error_code":"UDAPI1020","property_path":"getHistoricalCandleData.interval","invalid_value":"None"}]}
[E 240506 11:39:32 views:155] API request failed: Invalid filter: 'replace'
    Traceback (most recent call last):
      File "/Users/arshil.khan/Desktop/stock_comparison_project/stock_comparison_app/views.py", line 152, in generate_excel
        return render(request, 'comparison_result.html', {'comparison_data': comparison_data})
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/shortcuts.py", line 25, in render
        content = loader.render_to_string(template_name, context, request, using=using)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/loader.py", line 61, in render_to_string
        template = get_template(template_name, using=using)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/loader.py", line 15, in get_template
        return engine.get_template(template_name)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/backends/django.py", line 33, in get_template
        return Template(self.engine.get_template(template_name), self)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/engine.py", line 177, in get_template
        template, origin = self.find_template(template_name)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/engine.py", line 159, in find_template
        template = loader.get_template(name, skip=skip)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/loaders/cached.py", line 57, in get_template
        template = super().get_template(template_name, skip)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/loaders/base.py", line 28, in get_template
        return Template(
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 154, in __init__
        self.nodelist = self.compile_nodelist()
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 196, in compile_nodelist
        return parser.parse()
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 510, in parse
        raise self.error(token, e)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 508, in parse
        compiled_result = compile_func(self, token)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/defaulttags.py", line 861, in do_for
        nodelist_loop = parser.parse(
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 481, in parse
        raise self.error(token, e)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 479, in parse
        filter_expression = self.compile_filter(token.contents)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 597, in compile_filter
        return FilterExpression(token, self)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 693, in __init__
        filter_func = parser.find_filter(filter_name)
      File "/Users/arshil.khan/Desktop/datacomparison/venv/lib/python3.10/site-packages/django/template/base.py", line 603, in find_filter
        raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
    django.template.exceptions.TemplateSyntaxError: Invalid filter: 'replace'
[E 240506 11:41:19 smartConnect:243] Error occurred while making a POST request to https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData. Error: Something Went Wrong, Please Try After Sometime. URL: https://apiconnect.angelbroking.com/rest/secure/angelbroking/historical/v1/getCandleData, Headers: {'Content-type': 'application/json', 'X-ClientLocalIP': '127.0.0.1', 'X-ClientPublicIP': '106.193.147.98', 'X-MACAddress': '82:1a:a0:a5:f4:bc', 'Accept': 'application/json', 'X-PrivateKey': 'JDldPrre', 'X-UserType': 'USER', 'X-SourceID': 'WEB'}, Request: {'exchange': 'NSE', 'symboltoken': '3718', 'interval': 'ONE_MINUTE', 'fromdate': '2023-01-01 00:00', 'todate': '0003-01-02 15:29'}, Response: {'message': 'Something Went Wrong, Please Try After Sometime', 'errorcode': 'AB1004', 'status': False, 'data': None}
[E 240506 11:41:19 views:117] Error: 400 - {"status":"error","errors":[{"errorCode":"UDAPI1015","message":"to_date must be greater than or equal to from_date and Date should be in valid format: yyyy-mm-dd","propertyPath":"","invalidValue":{"fromDate":"2023-01-01","toDate":"0003-01-02"},"error_code":"UDAPI1015","property_path":"","invalid_value":{"fromDate":"2023-01-01","toDate":"0003-01-02"}}]}
